hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerProtocolServerSideTranslatorPB.java

```diff
@@ -99,7 +106,36 @@ public OMResponse submitRequest(RpcController controller,
           return submitRequestToRatis(request);
         }
       } else {
-        return submitRequestDirectlyToOM(request);
+        try {
+          OMClientRequest omClientRequest =
+              OzoneManagerRatisUtils.createClientRequest(request);
+          if (omClientRequest != null) {
+            request = omClientRequest.preExecute(ozoneManager);
+          } else {
+            // Still work is ongoing, so for some of the requests still
+            // following older approach.
+            return submitRequestDirectlyToOM(request);
+          }
+        } catch (IOException ex) {
+          // As some of the preExecute returns error. So handle here.
+          return createErrorResponse(request, ex);
+        }
+        OMClientRequest omClientRequest = OzoneManagerRatisUtils
+            .createClientRequest(request);
+
```

Comment:
Why do we need to create ClientRequest again here?

Reply:
This is because in preExecute, if the request is modified, we don't set it back.
I am planning to change this behavior in Volume Jira implementation.